src/flows/context/flow.query.tsx

```diff
@@ -278,6 +280,58 @@ export const FlowQueryProvider: FC = ({children}) => {
     status = RemoteDataState.Loading
   }
 
+  const queryDependents = (startID: string) => {
+    sendEvent(notebookQueryKey)
+    _queryDependents(startID)
+  }
+
+  const _queryDependents = (startID: string) => {
+    if (status === RemoteDataState.Loading) {
+      return
+    }
+
+    let map = generateMap(runMode === RunMode.Run)
+
+    if (!map.length) {
+      return
+    }
+
+    map = map.slice(map.indexOf(map.find(m => m.id === startID)))
+    event('Running Notebook QueryAll')
+
+    Promise.all(
+      map
+        .filter(q => !!q?.visual)
+        .map(stage => {
+          loading[stage.id] = RemoteDataState.Loading
+          setLoading(loading)
+          return query(stage.visual, stage.scope)
+            .then(response => {
+              loading[stage.id] = RemoteDataState.Done
+              setLoading(loading)
+              forceUpdate(stage.id, response)
+            })
+            .catch(e => {
+              forceUpdate(stage.id, {
+                error: e.message,
+              })
+              loading[stage.id] = RemoteDataState.Error
+              setLoading(loading)
+            })
```

Comment:
nit on the setLoading. I feel like we might find ourselves in a weird lifecycle place with this logic. is it possible to refactor this to use a `setLoading(prev => ())` approach to prevent potential weirdness?